packages/astro/src/runtime/server/index.ts

```diff
@@ -493,19 +493,19 @@ function createAstroGlobFn() {
 // Inside of getStaticPaths.
 export function createAstro(
 	filePathname: string,
-	_site: string,
+	_site: string | undefined,
 	projectRootStr: string
 ): AstroGlobalPartial {
-	const site = new URL(_site);
-	const url = new URL(filePathname, site);
+	const site = _site ? new URL(_site) : undefined;
+	const referenceURL = new URL(filePathname, `http://localhost`);
```

Comment:
Ah, I see you only grabbing the pathname below. Can you add a comment for why its okay to hardcode localhost here?